core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -99,6 +157,14 @@ public void setMessageCompression(boolean enable) {
   }
 
   @Override
+  public void setCompression(String compressorName) {
+    checkState(compressor == null, "compressor already set");
```

Comment:
Not being able to set this twice seem strange for a setter. It could be fine, but if that is the case it would have to be in the documentation.

I don't see why this isn't more precise:
```
checkState(!sendHeadersCalled, "sendHeaders has already been called");
```

In the case the user has already called sendHeaders it would produce a much better error message.